app/views/LoginView/UserForm.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { forwardRef, useImperativeHandle, useRef } from 'react';
+import { Keyboard, Text, View, TextInput as RNTextInput } from 'react-native';
+import { useDispatch } from 'react-redux';
+import { RouteProp, useNavigation, useRoute } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+
+import { loginRequest } from '../../actions/login';
+import Button from '../../containers/Button';
+import { ControlledFormTextInput } from '../../containers/TextInput';
+import I18n from '../../i18n';
+import { OutsideParamList } from '../../stacks/types';
+import { useTheme } from '../../theme';
+import sharedStyles from '../Styles';
+import UGCRules from '../../containers/UserGeneratedContentRules';
+import { useAppSelector } from '../../lib/hooks';
+import styles from './styles';
+
+interface ISubmit {
+	user: string;
+	password: string;
+}
+
+const schema = yup.object().shape({
+	user: yup.string().required(),
+	password: yup.string().required()
+});
+
+export interface UserFormRef {
+	getUser: () => string;
+}
+
+const UserForm = forwardRef<UserFormRef>((_, ref) => {
+	const passwordInput = useRef<RNTextInput | null>(null);
+
+	const {
+		params: { username }
+	} = useRoute<RouteProp<OutsideParamList, 'LoginView'>>();
+	const navigation = useNavigation<StackNavigationProp<OutsideParamList, 'LoginView'>>();
+
+	const {
+		control,
+		handleSubmit,
+		formState: { isValid },
+		getValues
+	} = useForm<ISubmit>({ mode: 'onChange', resolver: yupResolver(schema), defaultValues: { user: username || '' } });
+
+	const { colors } = useTheme();
+	const dispatch = useDispatch();
+
+	const getUser = () => getValues('user');
+	useImperativeHandle(ref, () => ({
+		getUser
+	}));
+
+	const {
+		Accounts_EmailOrUsernamePlaceholder,
+		Accounts_PasswordPlaceholder,
+		Accounts_PasswordReset,
+		Accounts_RegistrationForm_LinkReplacementText,
+		isFetching,
+		Accounts_RegistrationForm,
+		Site_Name,
+		inviteLinkToken
+	} = useAppSelector(state => ({
```

Comment:
When there is a need to get a lot of data, I have created custom hooks, to improve the code reading.